packages/valory/skills/decision_maker_abci/behaviours/sell_outcome_token.py

```diff
@@ -0,0 +1,215 @@
+            contract_address=self.params.conditional_tokens_address,
+            contract_public_id=ConditionalTokensContract.contract_id,
+            contract_callable="balanceOf",
+            data_key="data",
+            placeholder="outcome_token_balance",
+            id=self.current_condition_id,
+            owner=self.synchronized_data.safe_contract_address,
+        )
+        return response_msg
+
+    def _build_approval_tx(self) -> WaitableConditionType:
+        """Build an ERC20 approve transaction."""
+        response_msg = yield from self.get_contract_api_response(
+            performative=ContractApiMessage.Performative.GET_STATE,  # type: ignore
+            contract_address=self.collateral_token,
+            contract_id=str(ERC20.contract_id),
+            contract_callable="build_approval_tx",
+            spender=self.market_maker_contract_address,
+            amount=self.return_amount,
+        )
+
+        if response_msg.performative != ContractApiMessage.Performative.STATE:
+            self.context.logger.info(f"Could not build approval tx: {response_msg}")
+            return False
+
+        approval_data = response_msg.state.body.get("data")
+        if approval_data is None:
+            self.context.logger.info(f"Could not build approval tx: {response_msg}")
+            return False
+
+        batch = MultisendBatch(
+            to=self.market_maker_contract_address,
+            data=HexBytes(approval_data),
+        )
+        self.multisend_batches.append(batch)
+        return True
+
+    def _calc_sell_amount(self) -> WaitableConditionType:
+        """Calculate the sell amount of the conditional token."""
+        response_msg = yield from self.get_contract_api_response(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+            contract_address=self.market_maker_contract_address,
+            contract_id=str(FixedProductMarketMakerContract.contract_id),
+            contract_callable="calc_sell_amount",
+            return_amount=self.return_amount,
+            outcome_index=self.outcome_index,
+        )
+        if response_msg.performative != ContractApiMessage.Performative.RAW_TRANSACTION:
+            self.context.logger.error(
+                f"Could not calculate the sell amount: {response_msg}"
+            )
+            return False
+
+        sell_amount = response_msg.raw_transaction.body.get(
+            "outcomeTokenSellAmount", None
+        )
+        if sell_amount is None:
+            self.context.logger.error(
+                f"Something went wrong while trying to get the outcomeTokenSellAmount amount for the conditional token: {response_msg}"
+            )
+            return False
+
+        # TODO: not sure this is the correct use of remove_fraction_wei, is sell amount in wxDAI?
+        self.sell_amount = remove_fraction_wei(sell_amount, self.params.slippage)
```

Comment:
Yes, the sell amount should be in WEI, though I do not think we should subtract the slippage from it.